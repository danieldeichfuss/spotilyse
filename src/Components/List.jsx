import React from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";

export default function List({ items }) {
  const itemsToRender = items.map((item, i) => (
    <li className="List__item" key={i.toString()}>
      <span>{item.name}</span>
    </li>
  ));

  return (
    <ReactCSSTransitionGroup
      transitionName="AppearTransition"
      transitionAppear={true}
      transitionAppearTimeout={1000}
      transitionEnter={false}
      transitionLeave={false}
    >
      <ol className="List">{itemsToRender}</ol>
    </ReactCSSTransitionGroup>
  );
}
