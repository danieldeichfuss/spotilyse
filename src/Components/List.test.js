import React from "react";
import ReactDOM from "react-dom";
import List from "./List";
import renderer from "react-test-renderer";

const items = [];

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<List items={items} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// it("renders page correctly", () => {
//   const component = renderer.create(<List />);
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
