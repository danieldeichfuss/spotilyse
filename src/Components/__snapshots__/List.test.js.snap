// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page correctly 1`] = `
<CSSTransitionGroup
  transitionAppear={true}
  transitionAppearTimeout={1000}
  transitionEnter={false}
  transitionLeave={false}
  transitionName="AppearTransition"
>
  <ol
    className="List"
  />
</CSSTransitionGroup>
`;
